opencog/reasoning/RuleEngine/rules/pln/abduction.scm

```diff
@@ -0,0 +1,99 @@
+;==============================================================================
+; Abduction Rule
+;
+; AND(Inheritance A C, Inheritance B C) entails Inheritance A B
+;------------------------------------------------------------------------------
+
+(define pln-rule-abduction
+	(BindLink
+		(ListLink
+			(VariableNode "$A")
+			(VariableNode "$B")
+			(VariableNode "$C"))
+		(ImplicationLink
+			(AndLink
+				(InheritanceLink
+					(VariableNode "$A")
+					(VariableNode "$C"))
+				(InheritanceLink
+					(VariableNode "$B")
+					(VariableNode "$C"))
+				(VariableNode "$A")
+				(VariableNode "$B")
+				(VariableNode "$C"))
+			(ExecutionOutputLink
+				(GroundedSchemaNode "scm: pln-formula-abduction")
+				(ListLink
+					(InheritanceLink
+						(VariableNode "$A")
+						(VariableNode "$B"))
+					(InheritanceLink
+						(VariableNode "$A")
+						(VariableNode "$C"))
+					(InheritanceLink
+						(VariableNode "$B")
+						(VariableNode "$C"))
+					(VariableNode "$A")
+					(VariableNode "$B")
+					(VariableNode "$C"))))))
+; -----------------------------------------------------------------------------
+; Abduction Formula
+; -----------------------------------------------------------------------------
+
+; -----------------------------------------------------------------------------
+; Side-effect: TruthValue of AB may be updated
+; -----------------------------------------------------------------------------
+
+(define (pln-formula-abduction AB AC BC A B C)
+	(cog-set-tv!
+		AB
+		(pln-formula-abduction-side-effect-free
+			AB
+			AC
+			BC
+			A
+			B
+			C)))
+; -----------------------------------------------------------------------------
+; This version has no side effects and simply returns a TruthValue
+; -----------------------------------------------------------------------------
+
+(define (pln-formula-abduction-side-effect-free AB AC BC A B C)
```

Comment:
refer to https://github.com/opencog/opencog/issues/1401 on the updates to the deductionformula